options.go

```diff
@@ -0,0 +1,55 @@
+package cfdns
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/simplesurance/cfdns/logs"
+	"github.com/simplesurance/cfdns/logs/textlogger"
+	"golang.org/x/time/rate"
+)
+
+type Option func(*settings)
+
+type settings struct {
+	ratelim    *rate.Limiter
+	logger     *logs.Logger
+	httpClient *http.Client
```

Comment:
There are differences on the semantics. Currently the only way of configuring the client is by passing the option to the constructor, so after constructed the options never change. If I implemented the change you're suggesting, a caller would be able to do this:

```go
client := cfdns.New(...)

// then at any point I can change the client:
newopt := cfdns.WithHTTPClient(...)
newopt(client) // the client was modified
```

I don't want to allow that because of multiple reason, including possible race conditions. Because of that I think it is necessary that the options are applied on some object that won't be accessible to the caller later. This is only one possible approach.

Please take a look at the last version of the code. I embedded the settings struct on the client object. So the property I described above still holds, and the http client can now be accessed as `client.httpClient` instead of `client.cfg.httpClient`.

Reply:
Got it, never thought about that possibility.

nice solution :+1: